modules/common/src/Resource.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Drupal\common;
+
+use Procrastinator\HydratableTrait;
+use Procrastinator\JsonSerializeTrait;
+
+/**
+ * Class Resource.
```

Comment:
Add some documentation to explain more clearly the concept of a resource, relationship between versions and perspectives.